python/pyspark/sql/dataframe.py

```diff
@@ -422,6 +422,49 @@ def repartition(self, numPartitions):
         """
         return DataFrame(self._jdf.repartition(numPartitions), self.sql_ctx)
 
+    @since(1.6)
+    def repartitionBy(self, *cols, **kwargs):
```

Comment:
@davies @nongli Just curious, Should this api name keep consistent with scala? I mean, is that better to use `repartition` instead of  `repartitionBy`? Thus, to handle this part in the `def repartition(self, ... )` below.